internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -64,149 +64,167 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 	// Map of Gateway to the routes attached to it
 	gatewayRouteMap := make(map[string]sets.Set[string])
 
+	handledPolicies := make(map[types.NamespacedName]*egv1a1.BackendTrafficPolicy)
```

Comment:
curious why we need this map ? if its being used to handle duplicates, any idea how we ended up with duplicates ?

Reply:
For example:

There's a gateway with 3 different routes. You create a single policy resource that targets two of those three routes with two different `targetRef` elements.

This map is a way to prevent the policy resource from being added twice to the `res` array.

Reply:
re-reading, its because we have multiple targetRefs, can we execute `res = append(res, policy)` right after
```
for _, currPolicy := range backendTrafficPolicies {
    res = append(res, curPolicy)
    ....
```
and avoid extra memory

Reply:
It's not that simple. The code has two loops, the first handling the policies that target routes and the second handling the targets that target gateways.  Once we have multiple `targetRefs`, it's possible for the same policy to target both a route and a different gateway.

The second loop needs to work on the same deep copy of the policy as the first loop, otherwise the `status` section of the policy is not correctly updated. If we do as you suggest in the first loop, then the second loop would need to go over all of the content of `res` and find the correct policy instance if its there. The easy way to do this is like the code does right now.

Reply:
ah thanks for explaining
do we want to support a policy targeting a `Gateway` and `HTTPRoute` at the same time ? does our library gracefully handle this case when it comes to defaults and overrides ?

Reply:
It's no different than two different policies targeting both the gateway and the route. The plural `targetRefs` is just a form of syntactic sugar.